ui/raidboss/data/05-shb/eureka/delubrum_reginae_savage.js

```diff
@@ -596,6 +596,364 @@ export default {
+
+        // we need to filter for the three Avowed Avatars with the lowest IDs
+        // as they cast cleave at the different cardinals
+        const avatarOne = combatantDataAvatars.combatants.sort((a, b) => b.ID - a.ID).pop();
+        const avatarTwo = combatantDataAvatars.combatants.pop();
+        const avatarThree = combatantDataAvatars.combatants.pop();
+
+        const combatantPositions = {};
+        combatantPositions[getUnwaveringPosition(avatarOne)] = avatarOne;
+        combatantPositions[getUnwaveringPosition(avatarTwo)] = avatarTwo;
+        combatantPositions[getUnwaveringPosition(avatarThree)] = avatarThree;
+
+        // Avowed Avatars can spawn in the other positions
+        // Determine the location of Avowed Avatars
+        // North Avowed Avatar (-277, -97)
+        const northCombatant = combatantPositions[0];
+
+        // West Avowed Avatar (-277, -87)
+        const westCombatant = combatantPositions[3];
+
+        // South Avowed Avatar (-277, -77)
+        const southCombatant = combatantPositions[2];
+
+        // Get facings
+        const eastCombatantFacing = getFacing(eastCombatant);
+        const northCombatantFacing = getFacing(northCombatant);
+        const westCombatantFacing = getFacing(westCombatant);
+        const southCombatantFacing = getFacing(southCombatant);
+
+        // Get Blade of Entropy data
+        const eastCombatantBlade = data.blades[eastCombatant.ID];
+        const northCombatantBlade = data.blades[northCombatant.ID];
+        const westCombatantBlade = data.blades[westCombatant.ID];
+        const southCombatantBlade = data.blades[southCombatant.ID];
+
+        const bladeValues = {
+          '5942': 1,
+          '5943': -1,
+          '5946': 1,
+          '5947': -1,
+          '5956': 2,
+          '5957': -2,
+          '595A': 2,
+          '595B': -2,
+        };
+
+        // 1 = Right
+        // 0 = Left
+        const bladeSides = {
+          '5942': 1,
+          '5943': 1,
+          '5946': 0,
+          '5947': 0,
+          '5956': 1,
+          '5957': 1,
+          '595A': 0,
+          '595B': 0,
+        };
+
+        // Create map to improve readability of safeZone conditions
+        const dirNum = { north: 0, east: 1, south: 2, west: 3 };
+
+        // Only need to check cleaves from two parallel clones to determine safe spots
+        // because if the clone is cleaving inside, then we know where other clones
+        // are cleaving in order to make a '+' where the ends are each cleaved by one
+        // clone and the middle square is safe
+        let safeZone = null;
+        let adjacentZones = {};
+        if ((northCombatantFacing === dirNum.north && bladeSides[northCombatantBlade]) ||
+          (northCombatantFacing === dirNum.south && !bladeSides[northCombatantBlade])) {
+          // North clone cleaving inside east
+          safeZone = output.southwest();
+          adjacentZones = {
+            0: bladeValues[eastCombatantBlade],
```

Comment:
`0` here could also be `[dirNum.north]` for readability, here and elsewhere.

Reply:
oh that's how you do that as a key